### PR TITLE
fix: restrict `block_number == current_block_number - 1` in `BlockHashGadget` (for scroll)

### PR DESCRIPTION
**Pull request by silathdiir**
https://hackmd.io/@toghrulmaharram/ByO6HM6Oj

### CODE REVIEW

**Comment by lispc** [comment]
eh we plan to support both scroll l2 evm and original l1 evm in long term. So we need to add a feature flag call "scroll", when enabled, only restrict block_number == current_block_number - 1,  when disabled, original l1 evm logic

**Review by kunxian-xia — APPROVED** [review]

**Comment by lispc** [comment]
it is too complex...  what about this?

```

#[cfg(not(feature = "scroll"))]
const num_prev_block_allowed = 256;
#[cfg(feature = "scroll")]
const num_prev_block_allowed = 1;

let num_prev_block_allowed = 
let diff_lt = LtGadget::construct(
            cb,
            current_block_number.expr(),
            num_prev_block_allowed.expr() + from_bytes::expr(&block_number.cells),
        );
```

**Comment by silathdiir** [comment]
> it is too complex... what about this?
> 
> ```
> 
> #[cfg(not(feature = "scroll"))]
> const num_prev_block_allowed = 256;
> #[cfg(feature = "scroll")]
> const num_prev_block_allowed = 1;
> 
> let diff_lt = LtGadget::construct(
>             cb,
>             current_block_number.expr(),
>             num_prev_block_allowed.expr() + from_bytes::expr(&block_number.cells),
>         );
> ```

It seems that it doesn't satisfy `current_block_number == block_number + 1` (this LTGadget constrains `current_block_number < block_number + 1`?).

**Comment by lispc** [comment]
#[cfg(feature = "scroll")]
const num_prev_block_allowed = 2

**Comment by silathdiir** [comment]
> #[cfg(feature = "scroll")] const num_prev_block_allowed = 2

With `block_lt == 1` and `diff_lt == 1`?

**Comment by silathdiir** [comment]
Updated according to @lispc 's method. Thanks.

**Review by kunxian-xia — APPROVED** [review]

File: zkevm-circuits/Cargo.toml [hunk 1]
```diff
@@ -54,11 +54,12 @@ mock = { path = "../mock" }
 pretty_assertions = "1.0.0"
 
 [features]
-default = ["test", "onephase", "enable-sign-verify"]
+default = ["test", "onephase", "enable-sign-verify", "scroll"]
```
Comment (lispc) [inline zkevm-circuits/Cargo.toml:57]: don't put it into scroll.    we want the "default" be compatible with l1 evm

**Pull request merged**